src/ixdat/techniques/deconvolution.py

```diff
@@ -10,289 +9,308 @@
-        TODO: change the name kernel to something else
+        """
+        Generate an ECMSImpulseResponse from measurement.
 
         Args:
-            tspan (list): tspan over which to calculate the impulse response
-            tspan_bg (list): tspan of background to subtract. if list of tspans is
-                passed, will interpolate between the points using
+            mol (str): Molecule to calculate the impulse response of
+            measurement (ECMSMeasurement): Measurement including impulse response
+                measurment. Measurement needs to contain calibration data for mol
+                (either using ixdat native ECMSMeasurement.calibrate() or external
+                 package (eg siq's "quantifier"))
+            tspan (list): tspan over which to calculate the impulse response. Needs to
+                include zero. #TODO: verify this
+            tspan_bg (list): tspan of background to subtract. if list of tspans
+                (list of lists) is passed, will interpolate between the points using
                 calc_linear_background()
             norm (bool): If true the impulse response is normalized to its
                 area. Default is True.
             matrix (bool): If true the circulant matrix constructed from the
                 impulse reponse is returned. Default is False.
+
+            Additional keyword arguments are passed on to ECMSImpulseResponse.__init__
+        Return ECMSImpulseResponse
         """
-        if self.type == "measured":
-            if type(tspan_bg[0]) is not list:
-                t_kernel, kernel = self.data.grab_flux(
-                    mol=self.mol, tspan=tspan, tspan_bg=tspan_bg
-                )
-            else:
-                t_kernel, kernel_raw = self.data.grab_flux(mol=self.mol, tspan=tspan)
-                bg = calc_linear_background(t_kernel, kernel_raw, tspans=tspan_bg)
-                kernel = kernel_raw - bg
-            if norm:
-                area = np.trapz(kernel, t_kernel)
-                kernel = kernel / area
-            if matrix:
-                kernel = np.tile(kernel, (len(kernel), 1))
-                i = 1
-                while i < len(t_kernel):  # TODO: pythonize this?
-                    kernel[i] = np.concatenate((kernel[0][i:], kernel[0][:i]))
-                    i = i + 1
+        print("Generating `ECMSImpulseResponse` from measurement.")
+        if type(tspan_bg[0]) is not list:
+            t_kernel, kernel = measurement.grab_flux(
+                mol=mol, tspan=tspan, tspan_bg=tspan_bg
+            )
         else:
-            raise TechniqueError("Cannot calculate impulse response without data.")
-        return t_kernel, kernel
+            t_kernel, kernel_raw = measurement.grab_flux(mol=mol, tspan=tspan)
+            bg = calc_linear_background(t_kernel, kernel_raw, tspans=tspan_bg)
+            kernel = kernel_raw - bg
+        if norm:
+            area = np.trapezoid(kernel, t_kernel)
```

Comment:
This doesn't work with my version of numpy (1.24.3). I had to change it to `trapz` which I think works with all supported versions.

Reply:
see longer comment below. I changed it because of deprecation warning in numpy vs 2.0.0 - changed it back now. dont want to break backwards compatibility this drastically